### ISSUE TITLE
Modules cannot instantiaze the LocalizationService

### ISSUE THREAD

**Issue by bmhm** [fisharebest/webtrees#2656]
The instruction `$listController = new ListController(app(IndividualListService::class), app(LocalizationService::class));` works just fine in `app/Module/IndividualListModule.php` and others.

However, if modules try to call `app(LocallizationService::class)`, you will receive:
```
Unresolvable dependency resolving [Parameter #0 [  integer $id ]] in class Fisharebest\Webtrees\Tree …/vendor/illuminate/container/Container.php:975
```

**Comment by fisharebest** [comment]
The localization service depends on the language.
Languages are modules.
Therefore this object is not available until after all the modules have been booted.

Your options are

1) fetch this object from the container when it is needed `app(Localization::class)`.
2) type-hint it as a parameter to one of your `action` methods.

**Comment by bmhm** [comment]
I'll update the PHPdocs from webtrees which will then "fix" this bug.

**Comment by fisharebest** [comment]
This was fixed in 7ef421a4290b6b468944b845497d6139b0eddd3f

You can now type-hint this class in your constructor.

`LocalizationService` had a dependency on `LocaleInterface`.  This dependency has been removed.

Instead, some of the methods of `LocalizationService` now require a `$locale` parameter.

You can obtain this using `$locale $request->getAttribute('locale`).
